src/app/rosetta/lib/account.ml

```diff
@@ -441,7 +396,70 @@ module Balance = struct
               [ make_balance_amount
                   ~liquid_balance:(Unsigned.UInt64.of_int64 liquid_balance)
                   ~total_balance:(Unsigned.UInt64.of_int64 total_balance) ]
-          ; metadata }
+          ; metadata=None }
+      in
+      match req.block_identifier with
+      | None ->
+          (* First try via GraphQL but fallback to archive database (and then
+           * omit the nonce!) if there was an issue *)
+          (match%bind
+            env.gql
+              ?token_id:(Option.map token_id ~f:Unsigned.UInt64.to_string)
+              ~address ()
+          with
+          | `Failed e ->
+              [%log' warn env.logger] "/account/balance : GraphQL request failed, trying again via the archive database" ~metadata:[("error", Errors.erase e |> Rosetta_models.Error.to_yojson)];
```

Comment:
can this message mention that the GraphQL query is for the block identifier? 

It looks like the database is always consulted for the balance